src/main/java/org/jenkinsci/plugins/edithelp/EditHelpRootAction.java

```diff
@@ -0,0 +1,155 @@
+    public static void getListFiles(String str) {
+        File f = new File(str);
+        File[] mydr = f.listFiles();
+        if (mydr == null)
+            return;
+        for (File s : mydr) {
+            if (s.isFile()) {
+                listWithFileNames.add(s);
+            } else if (s.isDirectory()) {
+                getListFiles(s.getAbsolutePath());
+            }
+        }
+    }
+
+    //array for cache storage
+    Map<String, String> map = new TreeMap<String, String>();
+
+    //reading files into cache
+    public EditHelpRootAction() throws IOException, FileNotFoundException {
+        Jenkins jn = Jenkins.getInstance();
+        if (jn != null) {
+            File dirfile = jn.getRootDir();         
+            String dirName = null;         
+            if(dirfile != null) {
+                dirName = dirfile.toString()+"/helpmanager";
+            //check if the directory exists
+                File helpManagerFile = new File(dirName);
+
+                if(!helpManagerFile.exists())
+                    if (!helpManagerFile.mkdirs()) {
+                        System.out.print("unable create directory");
+                    }
+
+                //reading files from the directory
+                getListFiles(dirName);
+
+                //stream reading from files
+                for (File fil : listWithFileNames) {
+                    StringBuilder sb = new StringBuilder();
+                    BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(fil), "UTF8"));
+                    try {
+                        String s;
+                        while ((s = br.readLine()) != null) {
+                            sb.append(s);
+                            sb.append("\n");
+                        }
+                    } finally {
+                        br.close();
+                    }
+                    map.put(fil.getName(),sb.toString());
+                }
+            }            
+        }            
+    }
+
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return null;
+    }
+
+    @Override
+    public String getUrlName() {
+        return "helpmanager";
+    }
+
+    //copying from array
+    public String getMyString() {
+        //process get request
+        String class_name = Stapler.getCurrentRequest().getParameter("class");
+
+        if(map.containsKey(class_name+".html")){
+            return map.get(class_name+".html");
+        }else{
+            return null;
+        }
+    }
+
+    //save text area into file and array
+    public String getUpdateMyString() throws IOException, FileNotFoundException {
+        //process get request
+        String class_name = Stapler.getCurrentRequest().getParameter("class");
+        String updated_class_name = Stapler.getCurrentRequest().getParameter("textArea");
```

Comment:
Confusing variable name. Is it actually a <code>helpContent</code>. 

🐛 There is also lack of basic validation, which is important since wrong help may corrupt Jenkins layout and prevent future editing of the help from this WebUI.